src/metabase/driver/query_processor.clj

```diff
@@ -136,6 +149,23 @@
         ;; Add :rows_truncated if we've hit the limit so the UI can let the user know
         (= num-results results-limit) (assoc-in [:data :rows_truncated] results-limit)))))
 
+(defn- format-rows [{:keys [report-timezone]} rows]
+  (let [format-value #(if (u/is-temporal? %)
+                       (u/->ISO8601DateTime % report-timezone)
+                       %)]
+    (for [row rows]
+      (map format-value row))))
```

Comment:
I think it might be a little more readable to do
```clojure
(for [row rows
       v  row]
  (if (u/is-temporal? v)
    (u/->iso-8601-datetime v report-timezone)
    v))
```